apps/backend/src/mutations/TechniqueMutations.spec.ts

```diff
@@ -0,0 +1,117 @@
+      })
+    ).resolves.toHaveProperty('reason', 'INSUFFICIENT_PERMISSIONS');
+  });
+
+  test('A logged in user officer can create a technique', () => {
+    const techniqueToCreate = {
+      name: 'Dummy technique 1',
+      shortCode: 'technique_1',
+      description: 'Technique 1 description',
+    };
+
+    return expect(
+      techniqueMutations.create(dummyUserOfficerWithRole, techniqueToCreate)
+    ).resolves.toMatchObject({ id: 1, ...techniqueToCreate });
+  });
+
+  test('A logged in user officer can update a technique', () => {
+    const techniqueToUpdate = {
+      id: 1,
+      name: 'Dummy technique 1',
+      shortCode: 'technique_1',
+      description: 'Technique 1 description',
+    };
+
+    return expect(
+      techniqueMutations.update(dummyUserOfficerWithRole, techniqueToUpdate)
+    ).resolves.toStrictEqual({ ...techniqueToUpdate });
+  });
+
+  test('A user cannot update a technique', () => {
+    return expect(
+      techniqueMutations.update(dummyUserWithRole, {
+        id: 1,
+        name: 'Test technique',
+        shortCode: '2024-06-07',
+        description: 'Test technique description',
+      })
+    ).resolves.toHaveProperty('reason', 'INSUFFICIENT_PERMISSIONS');
+  });
+
+  test('A logged in user officer can delete a technique', () => {
+    return expect(
+      techniqueMutations.delete(dummyUserOfficerWithRole, {
+        id: 1,
+      })
+    ).resolves.toBe(dummyTechnique1);
+  });
+
+  test('A user cannot delete a technique', () => {
+    return expect(
+      techniqueMutations.delete(dummyUserWithRole, {
+        id: 1,
+      })
+    ).resolves.toHaveProperty('reason', 'INSUFFICIENT_PERMISSIONS');
+  });
+
+  test('A logged in user officer can assign instruments to a technique', async () => {
+    return expect(
+      techniqueMutations.assignInstrumentsToTechnique(
+        dummyUserOfficerWithRole,
+        {
+          instrumentIds: [1, 2],
+          techniqueId: 1,
+        }
+      )
+    ).resolves.toBe(true);
+  });
+
+  test('A user cannot assign instruments to a technique', async () => {
+    return expect(
+      techniqueMutations.assignInstrumentsToTechnique(dummyUserWithRole, {
+        instrumentIds: [1, 2],
+        techniqueId: 1,
+      })
+    ).resolves.toHaveProperty('reason', 'INSUFFICIENT_PERMISSIONS');
+  });
+
+  test('A logged in user officer can remove an assigned instrument from a technique', async () => {
+    return expect(
+      techniqueMutations.removeInstrumentFromTechnique(
+        dummyUserOfficerWithRole,
+        {
+          instrumentId: 1,
+          techniqueId: 1,
+        }
+      )
+    ).resolves.toBe(true);
+  });
+
+  test('A user cannot remove an assigned instrument from a technique', async () => {
+    return expect(
+      techniqueMutations.removeInstrumentFromTechnique(dummyUserWithRole, {
+        instrumentId: 1,
+        techniqueId: 1,
+      })
+    ).resolves.toHaveProperty('reason', 'INSUFFICIENT_PERMISSIONS');
+  });
+});
```

Comment:
Can we also add an additional test to test the Unique key constraint in InstrumentTechniques table?

Reply:
ex., Trying to add the same InstrumentId and techniqueId combination.

Reply:
Just to check, are you referring to the composite key in the `technique_has_instruments` table? If we do this as a backend test then I think this means we would only be testing the implementation of the mockup. And if we add this as an e2e test then it seems we would just be testing how postgres composite keys work. Or did you mean otherwise?